### PR TITLE
Fix texture leaks

### LINKED ISSUE #108: Texture 'leaking'

### PR DESCRIPTION
**Pull request by JamesHinshelwood**
Fixes #108 

Let me know if you need an explanation of how it works...
It's not the neatest but works for now

### CODE REVIEW

File: Scripts/Util/TextureManager.cs [hunk 1]
```diff
@@ -12,27 +15,30 @@ public static Texture PackTextures(Texture[] textures, out Rect2[] uvs)
...
+            uvs[i] = new Rect2((float)xPos / atlasImage.GetWidth(), 0, (float)images[0].GetWidth() / atlasImage.GetWidth(), 1);
 
             images[i].Lock();
-            for(int x = 0; x < images[i].GetWidth(); x++)
+            for(int x = -MARGIN; x < images[i].GetWidth() + MARGIN; x++)
             {
                 for(int y = 0; y < images[i].GetHeight(); y++)
                 {
-                    atlasImage.SetPixel((images[0].GetWidth() * i) + x, y, images[i].GetPixel(x, y));
+                    atlasImage.SetPixel(xPos + x, y, images[i].GetPixel((x + images[i].GetWidth()) % images[i].GetWidth(), y));
```
Comment (adam-kucz) [inline Scripts/Util/TextureManager.cs:31]: why SetPixel and iteration instead of BlitRect ?

**Comment by adam-kucz** [comment]
and that shouldn't be happening
image

File: Scripts/Util/TextureManager.cs (hunk 1 above)
Comment (JamesHinshelwood) [reply Scripts/Util/TextureManager.cs:31]: Didn't realise that was a thing - will change

**Comment by JamesHinshelwood** [comment]
Hmm I didn't get that error on mine, I'll try it with blit see if it fixes it

**Review by adam-kucz — APPROVED** [review]

**Pull request merged**